mito-ai/mito_ai/message_history.py

```diff
@@ -50,103 +96,266 @@ class GlobalMessageHistory:
+        with self._lock:
+            thread = self._chat_threads[thread_id]
+            thread.ai_optimized_history.append(ai_optimized_message)
+            thread.display_history.append(display_message)
+            self._update_last_interaction(thread)
+
+            if thread.name == "(New Chat)" and len(thread.display_history) >= 2:
+                # Retrieve first user and assistant messages from display_history
+                user_message = None
+                assistant_message = None
+                for msg in thread.display_history:
+                    if msg["role"] == "user" and user_message is None:
+                        user_message = msg["content"]
+                    elif msg["role"] == "assistant" and assistant_message is None:
+                        assistant_message = msg["content"]
+                    if user_message and assistant_message:
+                        break
+                if user_message and assistant_message:
+                    name_gen_input = (user_message, assistant_message)
+
+            # Save the updated thread to disk
+            self._save_thread_to_disk(thread)
+
+        # Outside the lock, await the name generation if needed
+        if name_gen_input:
+            new_name = await generate_short_chat_name(str(name_gen_input[0]), str(name_gen_input[1]), llm_provider)
+            with self._lock:
+                # Update the thread's name if still required
+                thread = self._chat_threads[thread_id]
+                if thread is not None and thread.name == "(New Chat)":
+                    thread.name = new_name
+                    self._save_thread_to_disk(thread)
 
     def truncate_histories(self, index: int) -> None:
+        """
+        For the newest thread, truncate messages at the given index.
+        """
+        with self._lock:
+            thread_id = self._get_newest_thread_id()
+            if not thread_id:
+                return
+            thread = self._chat_threads[thread_id]
+            thread.ai_optimized_history = thread.ai_optimized_history[:index]
+            thread.display_history = thread.display_history[:index]
+            self._update_last_interaction(thread)
+            self._save_thread_to_disk(thread)
+
+    def delete_thread(self, thread_id: str) -> bool:
+        """
+        Deletes a chat thread by its ID. Removes both the in-memory entry and the JSON file.
+        """
+        with self._lock:
+            if thread_id in self._chat_threads:
+                del self._chat_threads[thread_id]
+                path = os.path.join(self._chats_dir, f"{thread_id}.json")
+                if os.path.exists(path):
+                    try:
+                        os.remove(path)
+                        return True
+                    except Exception as e:
+                        print(f"Error deleting thread {thread_id}: {e}")
+            return False
+
+    def get_threads(self) -> List[ChatThreadItem]:
```

Comment:
I expect that the vast majority of chats are never going to be deleted. So maybe instead of sending all of the threads to the frontend, we should have some sort of cutoff? We only send the most recent 50 threads for example? 

To make this easier, what do you think about using the creation_ts as the as the ID? That way we can easily pull the most recent X chats without having to open every single chat file (of which I guess there could be 100's) and read the file contents.